### ISSUE TITLE
tiup cluster check also support Chrony as NTP implementation

### ISSUE THREAD

**Issue by yahonda** [pingcap/tiup#1696]
## Feature Request

"tiup cluster check" command to support `Chrony` as NTP implementation in addition to `ntpd`

**Why the featue is needed:**
Starting from Red Hat Enterprise 7 and Ubuntu 18.04, both Linux distributions added support Chrony as NTP as well as ntpd.

In some cases Chronyd is better than ntpd as described in these documentations.

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/system_administrators_guide/ch-configuring_ntp_using_the_chrony_suite
> chronyd can usually synchronize the clock faster and with better accuracy.

https://ubuntu.com/blog/ubuntu-bionic-using-chrony-to-configure-ntp
> Starting with Ubuntu Bionic, the choice for fast and accurate time synchronization is chrony.

**Describe alternatives you've considered:**
Use current `ntpd`.

**Teachability, Documentation, Adoption, Migration Strategy:**
It looks like `checkNTP` depends on `ntpd` implementation, does not look like supporting `chrony`.

https://github.com/pingcap/tiup/blob/c9251280eef33ca3f32571875dc37b039cb67b5e/pkg/cluster/operation/check.go#L194-L210

**Comment by yahonda** [comment]
cc @makocchi-git

**Comment by dveeden** [comment]
Some history on this: https://github.com/pingcap/docs/pull/5283

Isn't RHEL using Chrony starting with RHEL8 and not RHEL7?

**Comment by yahonda** [comment]
I have created a new EC2 instance using "Red Hat Enterprise Linux 7 with High Availability" at AWS Marketplace AMI. It installs and starts up `chrony` by default. 

```shell
[ec2-user@ip-172-31-43-69 ~]$ more /etc/redhat-release
Red Hat Enterprise Linux Server release 7.9 (Maipo)
```

```shell
[ec2-user@ip-172-31-43-69 ~]$ rpm -qa |grep ntp
fontpackages-filesystem-1.44-8.el7.noarch
[ec2-user@ip-172-31-43-69 ~]$ rpm -qa |grep chro
chrony-3.4-1.el7.x86_64
[ec2-user@ip-172-31-43-69 ~]$ sudo systemctl status chrony
Unit chrony.service could not be found.
[ec2-user@ip-172-31-43-69 ~]$ sudo systemctl status chronyd
● chronyd.service - NTP client/server
   Loaded: loaded (/usr/lib/systemd/system/chronyd.service; enabled; vendor preset: enabled)
   Active: active (running) since Wed 2022-01-05 01:21:38 UTC; 2min 19s ago
     Docs: man:chronyd(8)
           man:chrony.conf(5)
 Main PID: 624 (chronyd)
   CGroup: /system.slice/chronyd.service
           └─624 /usr/sbin/chronyd

Jan 05 01:21:37 localhost.localdomain systemd[1]: Starting NTP client/server...
Jan 05 01:21:38 localhost.localdomain chronyd[624]: chronyd version 3.4 starting (+CMDMON +NTP +REFCLOCK +RTC +PRIVD...EBUG)
Jan 05 01:21:38 localhost.localdomain systemd[1]: Started NTP client/server.
Jan 05 01:21:45 ip-172-31-43-69.us-east-2.compute.internal chronyd[624]: Selected source 169.254.169.123
Hint: Some lines were ellipsized, use -l to show in full.
[ec2-user@ip-172-31-43-69 ~]$
```

**Comment by makocchi-git** [comment]
tiup uses [tidb-insight](https://github.com/pingcap/tidb-insight) inside. So we need to implement tidb-insight for checking `chrony` just like ntp.
Fortunately, it's already done by the latest [commit](https://github.com/pingcap/tidb-insight/commit/8dad166491c2595b6916713e7f589b65ed95b232) 👍

**Comment by dveeden** [comment]
@yahonda from https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/system_administrators_guide/ch-configuring_ntp_using_the_chrony_suite :

"The chrony suite is installed by default on some versions of Red Hat Enterprise Linux 7"

This means our tools and documentation should include details about and support for Chrony. https://docs.pingcap.com/tidb/stable/check-before-deployment#check-and-install-the-ntp-service already included details about Chrony, no follow up needed for the docs.

**Comment by AstroProfundis** [comment]
Indeed, we have added support for `tidb-insight` recently, and we'll do some tests and adjust how `tiup-cluster` handles the output of `tidb-insight` collector in the `check` subcommand before we can make that a new release.

**Comment by yahonda** [comment]
Thanks for the update. I'm looking for the future version of tiup supporting Chrony implementation.
